src/plugins/release-monitor/index.js

```diff
@@ -0,0 +1,282 @@
+ * @returns {Promise} Resolves when the status is created on the PR
+ * @private
+ */
+function getAllCommitForPR(context, number) {
+    return context.github.pullRequests.getCommits(
+        context.repo({
+            number
+        })
+    );
+}
+
+/**
+ * Updates the status on all the provided prs
+ * @param {Object} context - probot context object
+ * @param {Array<Object>} prs - Collection of prs
+ * @param {boolean} isSuccess - to create a success status
+ * @returns {Promise} Resolves when the status is created on the PR
+ * @private
+ */
+async function createStatusOnPRs({ context, prs, isSuccess }) {
+    const statusFunc = isSuccess ? createSuccessPRStatus : createPendingPRStatus;
+
+    await Promise.all(prs.map(async pr => {
+        const allCommits = await getAllCommitForPR(context, pr.number);
+
+        await statusFunc(context, pluckLatestCommitSha(allCommits));
+    }));
+}
+
+/**
+ * Get all the commits for a PR
+ * @param {Object} context - probot context object
+ * @param {boolean} isSuccess - to create a success status
+ * @returns {Promise} Resolves when the status is created on the PR
+ * @private
+ */
+async function createStatusOnAllPRs({ context, isSuccess }) {
+    const { data: allOpenPrs } = await getAllOpenPRs(context);
+
+    return createStatusOnPRs({
+        context,
+        prs: allOpenPrs,
+        isSuccess
+    });
+}
+
+/**
+ * Returns collection of all the PRs which do not have fix or docs tag on it
+ * @param {Object} context - probot context object
+ * @returns {Promise} promise
+ * @private
+ */
+async function getNonSemverPatchPRs(context) {
+    const { data: allOpenPrs } = await getAllOpenPRs(context);
+
+    return allOpenPrs.reduce(async(previousPromise, pr) => {
+        const coll = await previousPromise;
+        const allCommits = await getAllCommitForPR(context, pr.number);
+
+        if (!isMessageValidForPatchRelease(getCommitMessageForPR(allCommits, pr))) {
+            coll.push(pr);
+        }
+
+        return coll;
+    }, Promise.resolve([]));
+}
+
+/**
+ * Release label is present
+ * @param {Array<Object>} labels - collection of label objects
+ * @returns {boolean} True if release label is present
+ * @private
+ */
+function hasReleaseLabel(labels) {
+    return labels.some(({ name }) => name === RELEASE_LABEL);
+}
+
+/**
+ * Post Release label is present
+ * @param {Array<Object>} labels - collection of label objects
+ * @returns {boolean} True if post release label is present
+ * @private
+ */
+function hasPostReleaseLabel(labels) {
+    return labels.some(({ name }) => name === POST_RELEASE_LABEL);
+}
+
+/**
+ * Handler for issue label event
+ * @param {Object} context - probot context object
+ * @returns {Promise} promise
+ * @private
+ */
+async function issueLabeledHandler(context) {
+
+    // check if the label is post-release and the same issue has release label
+    if (hasPostReleaseLabel([context.payload.label]) && hasReleaseLabel(context.payload.issue.labels)) {
```

Comment:
It seems like this is the only usage of `hasPostReleaseLabel`. Could `hasPostReleaseLabel` be refactored to accept a single label rather than an array?